pkgs/by-name/so/sourcegit/package.nix

```diff
@@ -0,0 +1,87 @@
+{
+  lib,
+  stdenv,
+  buildDotnetModule,
+  fetchFromGitHub,
+  dotnetCorePackages,
+  copyDesktopItems,
+  makeDesktopItem,
+  libicns,
+
+  git,
+  xdg-utils,
+}:
+
+let
+  binPath = lib.makeBinPath [
+    git
+    xdg-utils
+  ];
+in
+
+buildDotnetModule rec {
+  pname = "sourcegit";
+  version = "8.30.1";
+
+  src = fetchFromGitHub {
+    owner = "sourcegit-scm";
+    repo = "sourcegit";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-HlyCOumeB3Vc4p+92YM1JrqW1d4p8+cRwKyHbpjVD4w=";
+  };
+
+  patches = [ ./fix-darwin-git-path.patch ];
+
+  dotnet-sdk = dotnetCorePackages.sdk_8_0;
+  dotnet-runtime = dotnetCorePackages.runtime_8_0;
+
+  nugetDeps = ./deps.nix;
+
+  projectFile = [ "src/SourceGit.csproj" ];
+
+  executables = [ "SourceGit" ];
+
+  dotnetFlags = [
+    "-p:PublishAot=false" # untill https://github.com/NixOS/nixpkgs/issues/280923 is fixed
+  ];
+
+  nativeBuildInputs = lib.optionals stdenv.hostPlatform.isLinux [
+    copyDesktopItems
+    libicns
+  ];
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "SourceGit";
+      exec = "SourceGit";
+      icon = "SourceGit";
+      desktopName = "SourceGit";
+      terminal = false;
+      comment = meta.description;
+    })
+  ];
+
+  preInstall = ''
+    makeWrapperArgs+=(
+      --suffix PATH : ${binPath}
```

Comment:
```suggestion
      --suffix PATH : ${lib.makeBinPath [ git xdg-utils ]}
```

Reply:
nixfmt splits this into multiple lines, that's why I extracted it. It doesn't look terrible as a multiline, but it's not pretty.
If you think it's better I can move the `let in` above the wrapping logic.

Reply:
Also, what's your opinion about this being suffix vs prefix?

Reply:
I would recommend to not hack around the formatting by moving things into let ins and similar.

I would suffix it to always prefer local available tools or git with more functions or scripts around xdg-open for other environment.

Reply:
Thanks for the feedback.
I don't completely agree with not having formatting hacks: having a multi-line expression interpolated into a multi-line string is a bit difficult to parse with your head. Though in this case it's not unreadable, so I did went with it.